extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java

```diff
@@ -536,4 +542,37 @@ public static List<OidcRequestFilter> getMatchingOidcRequestFilters(Map<OidcEndp
         }
 
     }
+
+    public static Uni<HttpResponse<Buffer>> sendRequest(io.vertx.core.Vertx vertx, HttpRequest<Buffer> request,
+            boolean blockingDnsLookup) {
+        if (blockingDnsLookup) {
+            return sendRequest(new Vertx(vertx), request, true);
+        } else {
+            return request.send();
+        }
+    }
+
+    public static Uni<HttpResponse<Buffer>> sendRequest(Vertx vertx, HttpRequest<Buffer> request, boolean blockingDnsLookup) {
+        if (blockingDnsLookup) {
+            return vertx.executeBlocking(new Callable<Void>() {
```

Comment:
@michalvavrik Hey Michal, what I'm not sure if whether we should use Vertx's own worker pool or if we have something in Quarkus ? @cescoffier Can you please advise ?

Reply:
I read it in Vert.x community discussion here https://groups.google.com/g/vertx/c/FgI5sf-DAK0, it is fairly old topic, but I didn't find anything newer. If @cescoffier have better idea I'm happy to dig into it.

Reply:
@michalvavrik I was thinking of something like running it on the Quarkus Infrastructure pool, something like that, I'm just not sure what is the best recommendation in such cases;

Reply:
@michalvavrik Oh indeed, I see the Mail client uses the same technique, so it must be good, I'll wait for a bit for Clement to comment and will merge later, thanks

Reply:
@michalvavrik Sure, I'd just like to have a bit more clarity re using the `Infrastructure.getDefaultWorkerPool()`

Reply:
When using URLs, DNS lookups are blocking. If your DNS server is slow (and the lookup result is not cached), you can block the event loop. So, it's better to offload the lookup when this can happen. 

Vert.x provides an async DNS resolver, but I do not recommend it. DNS is complex, and users are likely using the JVM default behavior.

Reply:
@cescoffier Thanks, can you clarify please if using the `Infrastructure.getDefaultWorkerPool()` to run the blocking call can be more efficient or is it meant for different purposes ? Also we have this executor in the security, 
https://github.com/quarkusio/quarkus/blob/main/extensions/security/runtime/src/main/java/io/quarkus/security/runtime/IdentityProviderManagerCreator.java#L37

I wonder if we should use a single pool, such as the one provided by the security `IdentityProviderManagerCreator` for various blocking tasks in the security flows

Reply:
>  Also we have this executor in the security,

Just to be clear, we almost never use this executor in security. It is just default implementation for case someone is not relying on the Vert.x HTTP extension. And not using it solved some weird issues, I don't remember details but you can check the PR in which I added this executor and you will see. I think most of our users including all that are using OIDC are using Vert.x `executeBlocking` from here https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/VertxBlockingSecurityExecutor.java

Reply:
The strange issue was related to HTTP/2 and solved by `executeBlocking` https://github.com/quarkusio/quarkus/issues/34912

Reply:
BTW I didn't use that executor because this code is in OIDC commons and it does not require security (or it's SPI).

Reply:
@michalvavrik Sounds good, thanks